File: 01/py/01.py
#!/usr/bin/env python3

data = list(map(int, open('input').read().split()))

# part 1
print(sum(x//3-2 for x in data))

# part 2
f = lambda x: x+f(x//3-2) if x > 0 else 0
print(sum(f(x) - x for x in data))

File: 01/rb/01.rb
#!/usr/bin/env ruby

input = File.readlines('input').map &:to_i

# part 1
p input.map{|x| x/3-2 }.sum

# part 2
def f x; x > 0 ? x+f(x/3-2) : 0; end
p input.map{|x| f(x) - x }.sum
